shared/actions/chat.js

```diff
@@ -493,22 +495,41 @@ function * _deleteMessage (action: DeleteMessage): SagaGenerator<any, any> {
       break
   }
 
-  if (!messageID) throw new Error('No messageID for message delete')
   if (!conversationIDKey) throw new Error('No conversation for message delete')
 
-  // TODO: Use delete message RPC call when it is available
-  const clientHeader = yield call(_clientHeader, CommonMessageType.delete, conversationIDKey)
-  clientHeader.supersedes = messageID
-
-  yield call(localPostLocalNonblockRpcPromise, {
-    param: {
-      conversationID: keyToConversationID(conversationIDKey),
-      identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
-      msg: {
-        clientHeader,
+  if (messageID) {
+    // Deleting a server message.
+    // TODO: Use delete message RPC call when it is available
+    const clientHeader = yield call(_clientHeader, CommonMessageType.delete, conversationIDKey)
+    clientHeader.supersedes = messageID
+
+    yield call(localPostLocalNonblockRpcPromise, {
+      param: {
+        conversationID: keyToConversationID(conversationIDKey),
+        identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
+        msg: {
+          clientHeader,
+        },
       },
-    },
-  })
+    })
+  } else {
+    // Deleting a local outbox message.
+    if (message.messageState !== 'failed') throw new Error('Tried to delete a non-failed message')
+    const outboxID = message.outboxID
+    if (!outboxID) throw new Error('No outboxID for pending message delete')
+
+    yield call(localCancelPostRpcPromise, {
+      param: {
+        outboxID: keyToOutboxID(outboxID),
+      },
+    })
+    // It's deleted, but we don't get notified that the conversation now has
+    // one less outbox entry in it.  Gotta remove it from the store ourselves.
+    yield put(({
```

Comment:
Yeah, @mmaxim tells me it's true.

It's not surprising that we don't get a Gregor server-generated notification, because the Gregor server knows nothing about failed messages -- they only exist on the local machine.  The local service could send us one, but doesn't now, and it sounds like @mmaxim would prefer to just the the RPC success be our acknowledgment of the deletion.